### PR TITLE
Add agent allow list documentation to agent reference

### LINKED ISSUE #1608: Document agent allow list
## This is an issue with:

- [ ] Bug (site functionality or styling)
- [ ] Errata (fix needed for doc content)
- [ ] New content (guide wanted)
- [x] Update (Add missing or refresh existing content)
- [ ] Enhancement (add new site functionality)

## Expected Behavior
[Agent docs](https://docs.sensu.io/sensu-go/5.11/reference/agent/) contains some reference to the agent allow list with examples.

## Current Behavior
Feature will be introduced in 5.12

## Possible Solution
Users wish to limit the capabilities of the Agent in regards to the check/hook commands it can execute. Classic users used client “safe_mode”, limiting the Agent to locally defined checks. In Sensu Go, we have introduced the agent `--allow-list` flag which reads configuration from either yaml or json files (see https://github.com/sensu/sensu-go/pull/3130 for examples). The agent allow list is a allowlist of executables for checks and check hooks.

Note: Always use "allowlist" when describing this feature; don't use "whitelist".

## Context
https://github.com/sensu/sensu-go/pull/3130

### PR DESCRIPTION
**Pull request by rgeniesse**
## Description
Adds `allow-list` flag to agent reference.

## Motivation and Context
Closes #1608

## Remaining items
- [x] Add example `allow-list` configuration file.

### CODE REVIEW

File: content/sensu-go/5.12/files/allow-list.yaml [hunk 1]
```diff
@@ -0,0 +1,14 @@
+- exec: /usr/local/bin/hello.sh
```
Comment (nikkictl) [inline content/sensu-go/5.12/files/allow-list.yaml]: I wouldn't use these examples literally, maybe there is a better use case we can exemplify.

File: content/sensu-go/5.12/reference/agent.md [hunk 2]
```diff
@@ -685,6 +686,7 @@ Usage:
   sensu-agent start [flags]
 
 Flags:
+      --allow-list string               path to yaml or json file containing allowlist of check or hook commands the agent can execute
```
Comment (nikkictl) [inline content/sensu-go/5.12/reference/agent.md]: You can get the help usage for this flag by running `sensu-agent -h`:

```suggestion
      --allow-list string               path to agent execution allow list configuration file
```

File: content/sensu-go/5.12/reference/agent.md [hunk 3]
```diff
@@ -1151,6 +1165,37 @@ sensu-agent start --statsd-metrics-port 6125
...
+| exec |      |
+----------------------|------
+description           | The command to allow the Sensu agent to run as a check or a hook.
+required              | true
+type                  | Array map thing
+example               | {{< highlight shell >}}- exec: /usr/local/bin/hello.sh{{< /highlight >}}
+
+| sha515 |      |
+----------------------|------
+description           | The checksum of the command or hook.
```
Comment (nikkictl) [inline content/sensu-go/5.12/reference/agent.md]: I'm not sure if this is what you meant..

```suggestion
description           | The checksum of the check or hook command.
```

File: content/sensu-go/5.12/reference/agent.md [hunk 4]
```diff
@@ -1151,6 +1165,37 @@ sensu-agent start --statsd-metrics-port 6125
...
 statsd-metrics-port: 6125{{< /highlight >}}
 
+### Allow list configuration
+
+| exec |      |
+----------------------|------
+description           | The command to allow the Sensu agent to run as a check or a hook.
+required              | true
+type                  | Array map thing
+example               | {{< highlight shell >}}- exec: /usr/local/bin/hello.sh{{< /highlight >}}
```
Comment (nikkictl) [inline content/sensu-go/5.12/reference/agent.md]: Again, maybe use real world example here.

File: content/sensu-go/5.12/reference/agent.md [hunk 5]
```diff
@@ -1151,6 +1165,37 @@ sensu-agent start --statsd-metrics-port 6125
...
+
+| sha515 |      |
+----------------------|------
+description           | The checksum of the command or hook.
+required              | maybe
```
Comment (nikkictl) [inline content/sensu-go/5.12/reference/agent.md]:
```suggestion
required              | false
```

File: content/sensu-go/5.12/reference/agent.md [hunk 6]
```diff
@@ -1151,6 +1165,37 @@ sensu-agent start --statsd-metrics-port 6125
...
+type                  | String
+example               | {{< highlight shell >}}sha512: 4f926bf4328...{{< /highlight >}}
+
+| args |      |
+----------------------|------
+description           | Arguments that the command in exec is allowed to have.
+required              | false
```
Comment (nikkictl) [inline content/sensu-go/5.12/reference/agent.md]:
```suggestion
required              | true
```

File: content/sensu-go/5.12/reference/agent.md [hunk 7]
```diff
@@ -1151,6 +1165,37 @@ sensu-agent start --statsd-metrics-port 6125
...
+type                  | Array
+example               | {{< highlight shell >}}args: 
+-"-w 80"{{< /highlight >}}
+
+| enable_env |      |
+----------------------|------
+description           | Enable environment.
```
Comment (nikkictl) [inline content/sensu-go/5.12/reference/agent.md]:
```suggestion
description           | Enable environment variables.
```

File: content/sensu-go/5.12/reference/agent.md [hunk 8]
```diff
@@ -1151,6 +1165,37 @@ sensu-agent start --statsd-metrics-port 6125
...
+| sha515 |      |
+----------------------|------
+description           | The checksum of the command or hook.
+required              | maybe
+type                  | String
+example               | {{< highlight shell >}}sha512: 4f926bf4328...{{< /highlight >}}
+
+| args |      |
+----------------------|------
+description           | Arguments that the command in exec is allowed to have.
```
Comment (nikkictl) [inline content/sensu-go/5.12/reference/agent.md]:
```suggestion
description           | Arguments for the exec command.
```

File: content/sensu-go/5.12/reference/agent.md (hunk 4 above)
Comment (rgeniesse) [reply content/sensu-go/5.12/reference/agent.md]: Yeah, good idea. I'll come up with one.

**Review by cwjohnston — CHANGES_REQUESTED** [review]
Let's in-line the allow-list.yml as a YML/JSON switcher. Look at the asset reference doc for an example.

**Comment by rgeniesse** [comment]
@cwjohnston this should be all set now.

File: content/sensu-go/5.12/reference/agent.md [hunk 9]
```diff
@@ -1158,6 +1172,87 @@ sensu-agent start --statsd-metrics-port 6125
...
+| exec |      |
+----------------------|------
+description           | The command to allow the Sensu agent to run as a check or a hook.
+required              | true
+type                  | String
+example               | {{< highlight shell >}}"exec": "/usr/local/bin/check_memory.sh"{{< /highlight >}}
+
+| sha512 |      |
+----------------------|------
+description           | The checksum of the check or hook command.
```
Comment (cwjohnston) [inline content/sensu-go/5.12/reference/agent.md]: If I understand sensu/sensu-go#3130 correctly, this should be the checksum of the executable, not the command. cc @nikkixdev for confirmation

**Review by cwjohnston — CHANGES_REQUESTED** [review]

File: content/sensu-go/5.12/reference/agent.md (hunk 9 above)
Comment (echlebek) [reply content/sensu-go/5.12/reference/agent.md]: That's my understanding as well

**Review by cwjohnston — APPROVED** [review]

**Pull request merged**